fairlearn/metrics/_metric_frame.py

```diff
@@ -651,19 +866,39 @@ def difference(
                 "Unrecognised method '{0}' in difference() call".format(method)
             )
 
-        mf = self.by_group.copy()
-        # Can assume errors='coerce', else error would already have been raised in .group_min
-        # Fill all non-scalar values with NaN
-        if isinstance(mf, pd.Series):
-            mf = mf.map(lambda x: x if np.isscalar(x) else np.nan)
-        else:
-            mf = mf.applymap(lambda x: x if np.isscalar(x) else np.nan)
+        return self.__difference(self.by_group, subtrahend)
+
+    def _calc_ratio_overall(self, by_group_frame, overall_frame):
+        def ratio_sub_one(x):
```

Comment:
This is part of the bugfix mentioned for the aggregates. Basically, this makes everything use `ratio_sub_one()` rather than using `min(x, 1/x)` in some places.